src/map/mob.c

```diff
@@ -3257,16 +3258,7 @@ static int mob_summonslave(struct mob_data *md2, int *value, int amount, uint16
 		if (mob->db_checkid(data.class_) == 0)
 			continue;
 
-		short x;
-		short y;
-
-		if (map->search_freecell(&md2->bl, 0, &x, &y, MOB_SLAVEDISTANCE, MOB_SLAVEDISTANCE, 0) != 0) {
-			data.x = x;
-			data.y = y;
-		} else {
-			data.x = md2->bl.x;
-			data.y = md2->bl.y;
-		}
+		map->search_freecell(&md2->bl, 0, (int16 *)&data.x, (int16 *)&data.y, MOB_SLAVEDISTANCE, MOB_SLAVEDISTANCE, SFC_DEFAULT);
```

Comment:
I'd rather avoid this explicit cast. Although for the entire range of values that could be returned the variables have the same memory layout, this can hide useful warnings or errors in the event the type of `data.x` and `data.y` is changed in the future.

Using a pair of support variables here would feel like a more sensible choice to me.

Reply:
Done.